fetch.bs

```diff
@@ -1408,18 +1469,50 @@ for each associated <a for="fetch group">fetch record</a> whose
+
+<p>HTTP/2 allows connections to be reused for multiple origin servers, as described in
+<a href=https://tools.ietf.org/html/rfc7540#section-9.1.1>Section 9.1 of HTTP/2</a>. [[HTTP2]]
+
+<p>HTTPS connections can be reused for a request if the TLS certificate for the connection is
+valid for the host in the URI. This happens when the certificate has multiple "subjectAltName"
+attributes or names with wildcards, one of which is valid for the authority in the URI.
+
+<h4 id=obtain-connection>Obtaining a Connection</h4>
+
 <p>To <dfn export id=concept-connection-obtain for=connection>obtain a connection</dfn>, given an
 <var>origin</var> and <var>credentials</var>, run these steps:
 
 <ol>
- <li><p>If <a for=connection>connection pool</a> contains a
- <a>connection</a> whose <b>origin</b> is <var>origin</var> and
- <b>credentials</b> is <var>credentials</var>, return that
- <a>connection</a>.
+ <li><p>If <a for=connection>connection pool</a> contains a <a>connection</a>
+ whose <b>origin</b> is <var>origin</var>
+ or whose <b>origin</b> is authoritative for <var>origin</var>,
+ and <b>credentials</b> is <var>credentials</var>, return that <a>connection</a>.
 
  <li>
   <p>Let <var>connection</var> be the result of establishing an HTTP connection to
-  <var>origin</var>. [[!HTTP]] [[!HTTP-SEMANTICS]] [[!HTTP-COND]] [[!HTTP-CACHING]] [[!HTTP-AUTH]] [[!TLS]]
+  <var>origin</var>. [[!HTTP]] [[!HTTP2]] [[!HTTP-SEMANTICS]] [[!HTTP-COND]] [[!HTTP-CACHING]] [[!HTTP-AUTH]] [[!TLS]]
+
+  <p>If <var>credentials</var> is true and if the user agent supports <a for=/>Token Binding</a>,
+  propose the use of <a for=/>Token Binding</a> while setting up a TLS connection by sending the
+  highest supported <a for=/>Token Binding</a> protocol version and supported cryptographic
+  algorithms and parameters (the
+  <dfn export id=concept-token-binding-key-parameters>token-binding key parameters</dfn>) in a
+  <code>token_binding</code> Client Hello Extension, as described in
+  <a href="https://tools.ietf.org/html/draft-ietf-tokbind-negotiation#section-2">section 2</a>
+  of the Token Binding Negotiation spec [[!TOKBIND-NEGOTIATION]].
+  If Token Binding Negotiation succeeds, indicating client-server agreement on protocol version
+  and <a for=/>token-binding key parameters</a>, update metadata for the TLS connection with the
+  results of the negotiation.
+
+  <p class="note no-backref">
+  The user agent will use <a for=/>Token Binding</a> for any <a for=/>request</a> sent over
+  a TLS connection for which Token Binding Negotiation was successful.
+  Since <a for=/>Token Binding</a> is used only when <var>credentials</var> is true, such a
+  connection will not be pooled with connections that have <var>credentials</var> is
+  false. Also, <a for=/>Token Binding</a> is compatible with connection reuse - HTTP/2 requests to
+  different origin servers coalesced over the same connection will use different
+  <a for=/>token-binding key</a>s if needed.
```

Comment:
Why would tokbind be negotiated only when |credentials| is true?  there are use cases, of which webauthn is one (fido UAF is another), where the protocol running over HTTP is including a Token Binding ID in its (signed) messages, and whether or not cookies are also being used is a completely separate thing (they may not be (unlikely, but possible) and we shouldn't mess that up).